Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/checkins/checkout/ContactJournalEntryCreator.kt

```diff
@@ -0,0 +1,76 @@
+package de.rki.coronawarnapp.presencetracing.checkins.checkout
+
+import dagger.Reusable
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.eventregistration.checkins.split.splitByMidnightUTC
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDateUtc
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import org.joda.time.Seconds
+import timber.log.Timber
+import javax.inject.Inject
+import kotlin.math.roundToLong
+
+@Reusable
+class ContactJournalEntryCreator @Inject constructor(
+    private val diaryRepository: ContactDiaryRepository
+) {
+
+    suspend fun createEntry(checkIn: CheckIn) {
+        if (checkIn.createJournalEntry) {
+            Timber.d("Creating journal entry for %s", this)
+
+            // 1. Create location if missing
+            val traceLocationIdString = checkIn.traceLocationId.utf8()
```

Comment:
To prevent future bugs due to mistakes when comparing the traceLocationId please align this with the `traceLocationId` handling of the rest of the app.

i.e. use a ByteString everywhere outside the database, and for the database base64 encode.

```
interface ContactDiaryLocation : HasStableId {
    val locationId: Long
    var locationName: String
    val phoneNumber: String?
    val emailAddress: String?
    val traceLocationID: okio.ByteString? // (or use our TraceLocationId typealias)
}
```

then just compare the bytestrings directly, and when storing the tracelocationID in the contact diary table encode as base64 (`traceLocationId.base64()`).